doc/usage.md

```diff
@@ -0,0 +1,109 @@
+node. Otherwise you can start your own instance:
+
+- [Initialize PostgreSQL database](https://www.postgresql.org/docs/9.5/static/app-initdb.html)
+  (`-E SQL_ASCII` flag is recommended!)
+- [Start PostgreSQL database](https://www.postgresql.org/docs/9.5/static/app-postgres.html)
+
+## 1. Generate compilation databse
+If you want to parse a C++ project, you have to create a compilation database
+file (http://clang.llvm.org/docs/JSONCompilationDatabase.html).
+
+### Compilation database from CMake
+If the project uses CMake build system then you can create the compilation
+database by CMake with the option `CMAKE_EXPORT_COMPILE_COMMANDS`:
+
+```bash
+cmake .. -DCMAKE_EXPORT_COMPILE_COMMANDS=1
+```
+
+This command creates the `compile_commands.json` file, this is the compilation
+database.
+
+### Compilation database from CodeCompass
+CodeCompass can also be used for generating compilation database. This is a more
+general solution because it is not only for CMake projects. Suppose you have
+`hello.cpp` in the current working directory.
+
+```bash
+CodeCompass_logger compilation_commands.json "g++ main.cpp"
+```
+
+The `CodeCompass_logger` is provided in the `bin` directory in the CodeCompass
+installation. The first command line argument is the output file name and the
+second argument is the build command which compiles your project. This can be a
+simple compiler invocation or starting a build system.
+
+## 2. Parse a project
+For parsing a project with CodeCompass, the following command has to be emitted:
+
+```bash
+keepalive CodeCompass_parser -w <workspace> -n <name> -i <input1> -i <input2> -d <connection_string>
+```
+
+- **Workspace**: This is a directory where the parsed project and different
+  config and log files are located.
+- **Input**: Several inputs can be given with `-i` flags. An input can be a
+  directory or a compilation database. The plugins iterate these inputs and
+  decide if they can use it. For example the C++ parser will consume the given
+  compilation databases and the text search parser will consume the files under
+  the given directories. It is entirely up to the parser what it does with the
+  input parameters.
+- **Database**: The plugins can use an SQL database as storage. By the
+  connection string the user can give the location of a running database
+  system.
+
+:exclamation: The `keepalive` is a script in the `bin` directory of CodeCompass
+installation. This is used to keep CodeCompass alive if it crashes for some
+reason. This way a long parsing session restarts if an error happens. The
+CodeCompass plugins should handle themselves what to do when a parsing is
+stopped and restarted with an existing database, so the parsing can continue
+from where it stopped. **Furthermore it sets some environment variables which
+are necessary to start the `CodeCompass_parser` binary, so its usage is
+mandatory!**
+
+### Using example:
+
+```bash
+# Parse the prepared project
+keepalive CodeCompass_parser \
```

Comment:
The project name flag (`-n`) is also necessary.
Furthermore, I think this should be a real example like this:
```bash
# Parse an example C++ project. We suppose that the database is located on port 6250 on localhost
keepalive CodeCompass_parser \
  -d "pgsql:host=localhost;database=example;user=myusername;port=6250;" 
  -w ~/cc/workdir \
  -n example \
  -i ~/cc/projects/example/compile_commands.json \
  -j4
```